native/src/widget/text_input.rs

```diff
@@ -683,6 +683,30 @@ impl State {
     pub fn cursor(&self) -> Cursor {
         self.cursor
     }
+
+    /// Moves the [`Cursor`] of the [`TextInput`] to the front of the input text.
+    ///
+    /// [`Cursor`]: struct.Cursor.html
+    /// [`TextInput`]: struct.TextInput.html
+    pub fn move_cursor_to_front(&mut self) {
+        self.cursor.move_to(0);
+    }
+
+    /// Moves the [`Cursor`] of the [`TextInput`] to the end of the input text.
+    ///
+    /// [`Cursor`]: struct.Cursor.html
+    /// [`TextInput`]: struct.TextInput.html
+    pub fn move_cursor_to_end(&mut self, value: &String) {
+        self.cursor.move_to(value.len());
```

Comment:
Instead of asking for a string reference (which should be `&str`), I believe we should be able to use `usize::MAX` here. The `Cursor` API is already fault-tolerant (i.e. the read methods ask for a `Value` and clamp the indices accordingly).

Reply:
Sounds good, will do that.